control_panel_api/permissions.py

```diff
@@ -31,23 +28,13 @@ class S3BucketPermissions(IsSuperuser):
     pass
 
 
-class IsSelf(BasePermission):
+class IsSelfOrSuperuser(BasePermission):
 
     def has_permission(self, request, view):
```

Comment:
This should work.

I don't have enough experience with DRF permissions to tell if this is good or bad.

My only observation is that my understanding is that this method is supposed to determine if logged user can perform the class operations requested, while the `has_object_permission()` is to check if he can perform that operation on the specific instance of the user. This method is checking both things (as you it checks the `view.kwargs.get('pk')`).

Maybe having something like this would be more correct:

```python
def has_permission(self, request, view):
    if is_superuser(request.user):
        return True

    # Normal users can only update
    if request.method in ['PUT', 'PATCH']:
        return True


def has_object_permission(self, request, view, obj):
    if is_superuser(request.user):
        return True

    # otherwise normal users can potentially delete themselves
    if request.method not in ['GET', 'PUT', 'PATCH']:
        return False

    if hasattr(request.user, 'auth0_id'):
        return request.user == obj
```

I've seen something similar in the documentation.
Something I realised while writing this code is that we may want to check the request method to not allow normal users to perform non-`GET`/`PUT`/`PATCH`(e.g. so they can't somehow delete themselves).

This obviously is a bit longer so maybe have all the logic in `has_permission()` (even the one to check object permissions) is alright. As I said, I don't have enough experience with DRF permissions to know if this is bad or good.